File: Kotlin/app/src/main/kotlin/ca/jwolf/cubesolver/tools/ConfigVisualizer.kt
package ca.jwolf.cubesolver.tools

import ca.jwolf.cubesolver.datamodels.PuzzleSolution
import ca.jwolf.cubesolver.utils.ProgramUtils

class ConfigVisualizer: Tool {
    override fun toolName(): String {
        return "Config Visualizer"
    }

    override fun run(): Boolean {

        val config = ProgramUtils.getConfigFromUser()
        val solution = PuzzleSolution(config.cubeSize, config.getPuzzlePieceSymbolLookup())

        config.puzzlePieces.forEach{puzzlePiece ->
            puzzlePiece.snapToValidInsert(config.cubeSize)
            solution.insertPiece(puzzlePiece)
            print(solution.renderToString("Piece ${puzzlePiece.id}: ${puzzlePiece.description} has ${puzzlePiece.maxOrientation} unique orientation(s)"))
            solution.removePiece(puzzlePiece.id)

        }

        return true

    }
}

File: CSharp/CubeSolver/Tools/ConfigVisualizer.cs
using CubeSolver.Utils;
using NetCubeSolver;

namespace CubeSolver.Tools;

public class ConfigVisualizer: ITool
{
    public string Name { get; } = "Config Visualizer";
    public bool Run()
    {
        var config = ProgramUtils.GetConfigFromUser();

        var solution = new PuzzleSolution(config.CubeSize, config.PuzzlePieceSymbolLut);

        foreach (var piece in config.PuzzlePieces)
        {
            piece.SnapToValidInsert(config.CubeSize);
            solution.InsertPiece(piece);
            Console.Write(solution.RenderToString($"Piece {piece.Id}: {piece.Description} has {piece.MaxOrientation} unique orientation(s)"));
            solution.RemovePiece(piece.Id);
        }
        
        
        return true;

    }
}